### DISCUSSION TITLE
Static graph for whole dataset

### DISCUSSION THREAD

**Discussion by ralfriegel5** [pyg-team/pytorch_geometric#7217] [Q&A] [answered]
I want to solve a regression task with _n_ samples. The inputs are graph-structured. The graph does not change over the samples, i.e. it is a static graph for the whole dataset. In other words, only the node features change over the samples.

As each sample is supposed to be one `Data` instance, putting the graph into each `Data` sample seems to be inefficient and redundant. Is there a more efficient way to do this?

Many thanks!

**Comment by rusty1s** [comment, marked as answer]
You can also define your graph outside of `data` objects, e.g.
```python
class Dataset(...):
     def __init__(self, ...):
          self.edge_index = ...
          self.data, self.slices = torch.load(self.processed_paths[0])
      def __getitem__(self, idx):
          data = super().__getitem__(self, idx)
          data.edge_index = self.edge_index
          return data

**Reply by ralfriegel5** [reply]
very simple. great, thanks
